### ISSUE TITLE
"ListTables Missing required field: schema_name" regardless if schema provided or not

### ISSUE THREAD

**Issue by Limp-monkey** [open-metadata/OpenMetadata#15350]
**Affected module**
Ingestion Framework

**Describe the bug**
When trying to use (or test) the Databricks Unity Catalog connector, it fails with error:
`ListTables Missing required field: schema_name`

This is happening regardless if we put the schema name or not
![image](https://github.com/open-metadata/OpenMetadata/assets/45358832/a3af3a75-b214-42eb-918f-1782a8e72cef)
![image](https://github.com/open-metadata/OpenMetadata/assets/45358832/24e9a070-2cea-4a35-94f0-052ea08f66e8)
![image](https://github.com/open-metadata/OpenMetadata/assets/45358832/84692586-9650-463c-9023-7e9a3f68dd9c)

**To Reproduce**

1. Create Unity Catalog Connector
2. Start Databricks SQL Warehouse used in connection
3. Run test connection

**Expected behavior**
Connection to Unity Catalog should be established and metadata ingested

**Version:**

-     OS: AKS
-     Python version: openmetadata ingestion 1.3
-     OpenMetadata version: 1.3
-     OpenMetadata Ingestion package version: [e.g. openmetadata-ingestion[docker]==1.3]

**Additional context**

**Comment by Nemeczek** [comment]
The Unity Catalog connector still suffers from this issue. 
```
[2024-03-26T11:39:45.145+0000] {test_connections.py:216} INFO - failed=["'GetTables': This is a mandatory step and we won't be able to extract necessary metadata. Failed due to: ListTables Missing required field: schema_name"] success=["'CheckAccess': Pass", "'GetDatabases': Pass", "'GetSchemas': Pass", "'GetQueries': Pass"] warning=["'GetViews': This is a optional and the ingestion will continue to work as expected.Failed due to: ListTables Missing required field: schema_name"]
[2024-03-26T11:39:45.146+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/openmetadata_managed_apis/workflows/ingestion/common.py", line 206, in metadata_ingestion_workflow
    workflow = MetadataWorkflow.create(config)
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/workflow/ingestion.py", line 102, in create
    return cls(config)
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/workflow/ingestion.py", line 78, in __init__
    super().__init__(
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/workflow/base.py", line 106, in __init__
    self.post_init()
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/workflow/ingestion.py", line 97, in post_init
    self.set_steps()
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/workflow/metadata.py", line 35, in set_steps
    self.source = self._get_source()
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/workflow/metadata.py", line 60, in _get_source
    source: Source = source_class.create(self.config.source.dict(), self.metadata)
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/source/database/unitycatalog/metadata.py", line 115, in create
    return cls(config, metadata)
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/source/database/unitycatalog/metadata.py", line 98, in __init__
    self.test_connection()
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/source/database/database_service.py", line 583, in test_connection
    test_connection_fn(self.metadata, self.connection_obj, self.service_connection)
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/source/database/unitycatalog/connection.py", line 91, in test_connection
    test_connection_steps(
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/connections/test_connections.py", line 265, in test_connection_steps
    return timeout(timeout_seconds)(_test_connection_steps)(
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/utils/timeout.py", line 54, in inner
    result = fn(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/connections/test_connections.py", line 104, in _test_connection_steps
    _test_connection_steps_during_ingestion(steps=steps)
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/connections/test_connections.py", line 219, in _test_connection_steps_during_ingestion
    raise SourceConnectionException(
metadata.ingestion.connections.test_connections.SourceConnectionException: Some steps failed when testing the connection: [failed=["'GetTables': This is a mandatory step and we won't be able to extract necessary metadata. Failed due to: ListTables Missing required field: schema_name"] success=["'CheckAccess': Pass", "'GetDatabases': Pass", "'GetSchemas': Pass", "'GetQueries': Pass"] warning=["'GetViews': This is a optional and the ingestion will continue to work as expected.Failed due to: ListTables Missing required field: schema_name"]]
```

**Comment by Nemeczek** [comment]
Hey, 

we managed to get into bottom of this.
Based on the file: ingestion\src\metadata\ingestion\source\database\unitycatalog\connection.py 
we deducted that Test Connection tests of first Catalog found in Unity Catalog. So I issued the request to list catalogs `api/2.1/unity-catalog/catalogs`: 

And the first one was:
```
        {
            "name": "__databricks_internal",
            "owner": "System user",
            "comment": "Databricks internal catalog (auto-created)",
            "catalog_type": "INTERNAL_CATALOG",
```

which per @ulixius9 suggestion does not contain data - or probably it is visible only to databricks itself. 
After adding dummy catalog called `__a` the Test Connection and the Pipeline actually works.
